packages/@ngtools/webpack/src/extract_i18n_plugin.ts

```diff
@@ -0,0 +1,165 @@
+    } catch (err) {
+      throw new Error(`An error happened while parsing ${this._tsConfigPath} JSON: ${err}.`);
+    }
+    const tsConfig = ts.parseJsonConfigFileContent(
+      tsConfigJson, ts.sys, basePath, null, this._tsConfigPath);
+
+    let fileNames = tsConfig.fileNames;
+    if (options.hasOwnProperty('exclude')) {
+      let exclude: string[] = typeof options.exclude == 'string'
+        ? [options.exclude as string] : (options.exclude as string[]);
+
+      exclude.forEach((pattern: string) => {
+        const basePathPattern = '(' + basePath.replace(/\\/g, '/')
+            .replace(/[\-\[\]\/{}()+?.\\^$|*]/g, '\\$&') + ')?';
+        pattern = pattern
+        // Replace windows path separators with forward slashes.
+          .replace(/\\/g, '/')
+          // Escape characters that are used normally in regexes, except stars.
+          .replace(/[\-\[\]{}()+?.\\^$|]/g, '\\$&')
+          // Two stars replacement.
+          .replace(/\*\*/g, '(?:.*)')
+          // One star replacement.
+          .replace(/\*/g, '(?:[^/]*)')
+          // Escape characters from the basePath and make sure it's forward slashes.
+          .replace(/^/, basePathPattern);
+
+        const re = new RegExp('^' + pattern + '$');
+        fileNames = fileNames.filter(x => !x.replace(/\\/g, '/').match(re));
+      });
+    } else {
+      fileNames = fileNames.filter(fileName => !/\.spec\.ts$/.test(fileName));
+    }
+    this._rootFilePath = fileNames;
+
+    // By default messages will be generated in basePath
+    let genDir = basePath;
+
+    if (options.hasOwnProperty('genDir')) {
+      genDir = path.resolve(process.cwd(), options.genDir);
+    }
+
+    this._compilerOptions = tsConfig.options;
+    this._angularCompilerOptions = Object.assign(
+      { genDir },
+      this._compilerOptions,
+      tsConfig.raw['angularCompilerOptions'],
+      { basePath }
+    );
+
+    this._basePath = basePath;
+    this._genDir = genDir;
+
+    // this._compilerHost = new WebpackCompilerHost(this._compilerOptions, this._basePath);
+    this._compilerHost = ts.createCompilerHost(this._compilerOptions, true);
+    this._program = ts.createProgram(
+      this._rootFilePath, this._compilerOptions, this._compilerHost);
+
+    if (options.hasOwnProperty('i18nFormat')) {
+      this._i18nFormat = options.i18nFormat;
+    }
+  }
+
+  apply(compiler: any) {
+    this._compiler = compiler;
+
+    compiler.plugin('make', (compilation: any, cb: any) => this._make(compilation, cb));
+
+    compiler.plugin('after-emit', (compilation: any, cb: any) => {
+      this._donePromise = null;
+      this._compilation = null;
+      compilation._ngToolsWebpackXi18nPluginInstance = null;
+      cb();
+    });
+  }
+
+  private _make(compilation: any, cb: (err?: any, request?: any) => void) {
+    this._compilation = compilation;
+    if (this._compilation._ngToolsWebpackXi18nPluginInstance) {
+      return cb(new Error('An @ngtools/webpack xi18n plugin already exist for ' +
```

Comment:
Changes pushed. Travis build failed though. I don't understand why it failed, can you take a look and let me know if I broke something.